opensrp-immunization/src/main/java/org/smartregister/immunization/view/ServiceCard.java

```diff
@@ -205,11 +211,16 @@ private void updateStateUi() {
                 nameTV.setText(getServiceName() + " - " + DATE_FORMAT.format(getDateDone()));
                 break;
             case OVERDUE:
-                setBackgroundResource(R.drawable.vaccine_card_background_red);
+                if (ImmunizationLibrary.getInstance().isHideVaccineOverdueStatus()) {
+                    setBackgroundResource(R.drawable.vaccine_card_background_white);
+                    nameTV.setTextColor(context.getResources().getColor(R.color.silver));
```

Comment:
Because background is set to white here, text color will not be visible if it isn't set to silver.

You can also check line number 206 and 233. In these both switch cases whenever the background color is changed to white, the `nameTV` is always changed to `R.color.silver`. I followed the same pattern

Reply:
Do correct me if I am confusing something here